### PR TITLE
MINOR: [Release] Add RC number and version to verify release candidate success message

### PR DESCRIPTION
**Pull request by raulcd**
I've tested locally when verifying 10.0.1 rc 0:
```
Release candidate number: 0 for version: 10.0.1 looks good!
```

### CODE REVIEW

File: dev/release/verify-release-candidate.sh [hunk 1]
```diff
@@ -1198,5 +1198,5 @@ test_binary_distribution
 
 TEST_SUCCESS=yes
 
-echo 'Release candidate looks good!'
+echo "Release candidate number: ${RC_NUMBER} for version: ${VERSION} looks good!"
```
Comment (kou) [inline dev/release/verify-release-candidate.sh:1201]: It's not a strong opinion but how about simplifying it?

```suggestion
echo "Release candidate ${VERSION}-RC${RC_NUMBER} looks good!"
```

**Review by kou — APPROVED** [review]
+1

**Comment by raulcd** [comment]
@kou I think this can be merged :)

**Comment by kou** [comment]
Merged!

**Pull request merged**